src/relstorage/cache/storage_cache.py

```diff
@@ -772,11 +773,63 @@ def after_poll(self, cursor, prev_tid_int, new_tid_int, changes):
 
             self.__poll_replace_checkpoints(cursor, global_checkpoints, new_tid_int)
 
-        if not global_checkpoints_in_future and len(self.delta_after0) >= self.delta_size_limit:
+        if not global_checkpoints_in_future and self._should_suggest_shifted_checkpoints():
             self._suggest_shifted_checkpoints()
 
         return changes
 
+    #: By default, a 70% chance when we're full.
+    CP_REPLACEMENT_CHANCE_WHEN_FULL = float(
+        os.environ.get(
+            'RELSTORAGE_CP_REPLACEMENT_CHANCE_WHEN_FULL',
+            "0.7"
+        )
+    )
+
+    #: If we're just close, a 20% chance.
+    CP_REPLACEMENT_CHANCE_WHEN_CLOSE = float(
+        os.environ.get(
+            'RELSTORAGE_CP_REPLACEMENT_CHANCE_WHEN_CLOSE',
+            "0.2"
+        )
+    )
+
+    #: Start considering that we're close when we're 80% full.
+    CP_REPLACEMENT_BEGIN_CONSIDERING_PERCENT = float(
+        os.environ.get(
+            'RELSTORAGE_CP_REPLACEMENT_BEGIN_CONSIDERING_PERCENT',
+            "0.8"
+        )
+    )
+
+    def _should_suggest_shifted_checkpoints(self, _random=random.random):
```

Comment:
The old behaviour can get be had by setting 

`RELSTORAGE_CP_REPLACEMENT_CHANCE_WHEN_FULL = 1`
`RELSTORAGE_CP_REPLACEMENT_CHANCE_WHEN_CLOSE = 0`
`RELSTORAGE_CP_REPLACEMENT_BEGIN_CONSIDERING_PERCENT = 0`

?

Reply:
Pretty much. `RELSTORAGE_CP_REPLACEMENT_BEGIN_CONSIDERING_PERCENT` should be 1 for fastest evaluation, but 0 gets the job done.